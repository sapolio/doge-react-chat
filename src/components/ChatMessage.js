import React from 'react';
import classnames from 'classnames';
import { withStyles } from 'material-ui';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';
import Avatar from './Avatar';

const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    'flex-direction': 'row-reverse',
  },
  message: {
    maxWidth: '70%',
    minWidth: '10%',
    padding: theme.spacing.unit,
    marginLeft: theme.spacing.unit * 2,
    backgroundColor: '#fed'
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#def'
  },
});

const ChatMessage = ({ classes, sender, content }) => {
  const isMessageFromMe = sender === 'me';

 
  return (
    <div className={classnames(
      classes.messageWrapper,
      isMessageFromMe && classes.messageWrappperFromMe
    )}>
      <Avatar title={sender} />
      <Paper className={classnames(
        classes.message,
        isMessageFromMe && classes.messageFromMe
      )}>
        <Typography variant="caption">
          {sender}
        </Typography>
        <Typography variant="body1">
          {content}
        </Typography>
      </Paper>
    </div>
  );
};

export default withStyles(styles)(ChatMessage);
