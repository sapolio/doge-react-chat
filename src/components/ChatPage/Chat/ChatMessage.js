import React from 'react';
import classnames from 'classnames';
import { withStyles } from 'material-ui';
import Typography from 'material-ui/Typography';
import Paper from 'material-ui/Paper';
import moment from 'moment';
import Avatar from '../Avatar';
import senderName from '../../../utils/senderName';
import colorFrom from '../../../utils/colors';


const styles = theme => ({
  messageWrapper: {
    display: 'flex',
    justifyContent: 'flex-start',
    alignItems: 'center',
    padding: `${theme.spacing.unit}px ${theme.spacing.unit * 3}px`,
  },
  messageWrappperFromMe: {
    'flex-direction': 'row-reverse',
  },
  statusWrapper: {
    'justify-content': 'center',
  },
  message: {
    maxWidth: '70%',
    minWidth: '10%',
    padding: theme.spacing.unit,
    marginLeft: theme.spacing.unit * 2,
    backgroundColor: '#fee'
  },
  messageFromMe: {
    marginRight: theme.spacing.unit * 2,
    backgroundColor: '#def'
  },
});

const ChatMessage = ({ classes, sender, content, fromMe, statusMessage, createdAt }) => {

  const displayedName = senderName(sender);

  if (statusMessage) {
    return (
      <div className={classnames(classes.messageWrapper, classes.statusWrapper)}>
        <Typography>
          <Typography variant="caption" style={{ color: colorFrom(sender._id)}} className={classes.statusMessageUser}>
            {displayedName}
          </Typography>
          {content}
          <Typography variant="caption" component="span">
            {moment(createdAt).fromNow()}
          </Typography>
        </Typography>
      </div>
    )
  }

  return (
    <div className={classnames(
      classes.messageWrapper,
      fromMe && classes.messageWrappperFromMe
    )}>
      <Avatar title={sender.username} />
      <Paper className={classnames(
        classes.message,
        fromMe && classes.messageFromMe
      )}>
        <Typography variant="caption">
          {sender.username}
        </Typography>
        <Typography variant="body1">
          {content}
        </Typography>
      </Paper>
    </div>
  );
};

export default withStyles(styles)(ChatMessage);
