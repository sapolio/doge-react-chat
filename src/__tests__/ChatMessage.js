/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatMessage from '../components/ChatPage/Chat/ChatMessage';

jest.mock('moment', () => () => ({ fromNow: () => 'string' }));

describe('<ChatMessage />', () => {
  const props = {
    sender: {
      _id: '123456789',
      firstName: 'John',
      lastName: 'Doe',
      username: 'user',
    },
    content: 'string',
    fromMe: false,
    statusMessage: false,
    createdAt: 'string',
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ChatMessage {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('renders status message', () => {
    const tree = renderer.create(<ChatMessage {...props} statusMessage />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders not from me', () => {
    const tree = renderer.create(<ChatMessage {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders from me', () => {
    const tree = renderer.create(<ChatMessage {...props} fromMe />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders username only', () => {
    const tree = renderer
      .create(<ChatMessage
        {...props}
        sender={{
            _id: '123456789',
            username: 'user',
          }}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
