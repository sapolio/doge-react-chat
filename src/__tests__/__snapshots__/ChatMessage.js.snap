// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessage /> renders from me 1`] = `
<div
  className="ChatMessage-messageWrapper-60 ChatMessage-messageWrappperFromMe-61"
>
  <div
    className="MuiAvatar-root-89 MuiAvatar-colorDefault-90"
    style={
      Object {
        "backgroundColor": "#ff9800",
      }
    }
  >
    JD
  </div>
  <div
    className="MuiPaper-root-92 MuiPaper-shadow2-96 MuiPaper-rounded-93 ChatMessage-message-63 ChatMessage-messageFromMe-64"
  >
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
      style={
        Object {
          "color": "#ff9800",
        }
      }
    >
      John Doe
    </span>
    <p
      className="MuiTypography-root-65 MuiTypography-body1-74"
    >
      string
    </p>
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
    >
      string
    </span>
  </div>
</div>
`;

exports[`<ChatMessage /> renders not from me 1`] = `
<div
  className="ChatMessage-messageWrapper-60"
>
  <div
    className="MuiAvatar-root-89 MuiAvatar-colorDefault-90"
    style={
      Object {
        "backgroundColor": "#ff9800",
      }
    }
  >
    JD
  </div>
  <div
    className="MuiPaper-root-92 MuiPaper-shadow2-96 MuiPaper-rounded-93 ChatMessage-message-63"
  >
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
      style={
        Object {
          "color": "#ff9800",
        }
      }
    >
      John Doe
    </span>
    <p
      className="MuiTypography-root-65 MuiTypography-body1-74"
    >
      string
    </p>
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
    >
      string
    </span>
  </div>
</div>
`;

exports[`<ChatMessage /> renders status message 1`] = `
<div
  className="ChatMessage-messageWrapper-60 ChatMessage-statusWrapper-62"
>
  <p
    className="MuiTypography-root-65 MuiTypography-body1-74"
  >
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
      style={
        Object {
          "color": "#ff9800",
        }
      }
    >
      John Doe
    </span>
    string
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
    >
      string
    </span>
  </p>
</div>
`;

exports[`<ChatMessage /> renders username only 1`] = `
<div
  className="ChatMessage-messageWrapper-60"
>
  <div
    className="MuiAvatar-root-89 MuiAvatar-colorDefault-90"
    style={
      Object {
        "backgroundColor": "#ff9800",
      }
    }
  >
    U
  </div>
  <div
    className="MuiPaper-root-92 MuiPaper-shadow2-96 MuiPaper-rounded-93 ChatMessage-message-63"
  >
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
      style={
        Object {
          "color": "#ff9800",
        }
      }
    >
      user
    </span>
    <p
      className="MuiTypography-root-65 MuiTypography-body1-74"
    >
      string
    </p>
    <span
      className="MuiTypography-root-65 MuiTypography-caption-75"
    >
      string
    </span>
  </div>
</div>
`;
